src/v2/style-guide/index.md

```diff
@@ -0,0 +1,1805 @@
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+```
+components/
+|- MyComponent.vue
+```
+
+```
+components/
+|- my-component.vue
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Base component names <sup data-p="b">strongly recommended</sup>
+
+**Base components (a.k.a. presentational, dumb, or pure components) that apply app-specific styling and conventions should all begin with a specific prefix, such as `Base`, `App`, or `V`.**
+
+{% raw %}
+<details>
+<summary>
+  <h4>Detailed Explanation</h4>
+</summary>
+{% endraw %}
+
+These components lay the foundation for consistent styling and behavior in your application. They can only contain:
+
+- HTML elements,
+- other `Base`-prefixed components, and
+- 3rd-party UI components.
+
+They'll never contain:
+
+- global state (e.g. from a Vuex store)
+-
+
+Their names often include the name of an element they wrap (e.g. `BaseButton`, `BaseTable`), unless no element exists for their specific purpose (e.g. `BaseIcon`). If you build similar components for a more specific context, they will almost always consume these components (e.g. `BaseButton` may be used in `ButtonSubmit`).
+
+Some advantages of this convention:
+
+- When organized alphabetically in editors, your app's base components are all listed together, making them easier to identify.
+
+- Since component names should always be multi-word, this convention prevents you from having to choose an arbitrary prefix for simple component wrappers (e.g. `MyButton`, `VueButton`).
+
+- Since these components are so frequently used, you may want to simply make them global instead of importing them everywhere. A prefix makes this possible with Webpack:
+
+  ``` js
+  var requireComponent = require.context("./src", true, /^Base[A-Z]/)
+  requireComponent.keys().forEach(function (fileName) {
+    var baseComponentConfig = requireComponent(fileName)
+    baseComponentConfig = baseComponentConfig.default || baseComponentConfig
+    var baseComponentName = baseComponentConfig.name || (
+      fileName
+        .replace(/^.+\//, '')
+        .replace(/\.\w+$/, '')
+    )
+    Vue.component(baseComponentName, baseComponentConfig)
+  })
+  ```
+
+{% raw %}</details>{% endraw %}
+
+{% raw %}<div class="style-example example-bad">{% endraw %}
+#### Bad
+
+```
+components/
+|- MyButton.vue
+|- VueTable.vue
+|- Icon.vue
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+```
+components/
+|- BaseButton.vue
+|- BaseTable.vue
+|- BaseIcon.vue
+```
+
+```
+components/
+|- AppButton.vue
+|- AppTable.vue
+|- AppIcon.vue
+```
+
+```
+components/
+|- VButton.vue
+|- VTable.vue
+|- VIcon.vue
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Single-instance component names <sup data-p="b">strongly recommended</sup>
+
+**Components that should only ever have a single active instance should begin with the `The` prefix, to denote that there can be only one.**
```

Comment:
:/ I don't know about this. `The` seems weird to me, especially when used as a tag. I often go with something like `App` or `Site` instead, e.g. `AppFooter` or `SiteHeader`.

Reply:
`The` is the only one that unequivocally implies a single instance though. I've seen the `App` prefix commonly used for simple element wrappers, for example, like `AppButton`.

Reply:
Hm… when I say "the avatar," I don't mean there's only one avatar in the whole app right? Maybe it's _the_ avatar on the user profile? I don't know, this recommendation isn't very convincing to me TBH (but again maybe it's just me, as I'm not a native speaker).

Reply:
That could be true, but we do have an international userbase, so that's not a reason to dismiss what you're saying. Is it the specific prefix or the idea of the convention in general that's not convincing you?